### ISSUE TITLE
Using functional bean defs in AWS Lambda I get: Caused by: java.lang.ClassNotFoundException: org.springframework.cloud.function.web.source.DestinationResolver

### ISSUE THREAD

**Issue by adolfo-cia** [spring-cloud/spring-cloud-function#784]
**Describe the bug**
First I don't know if this is a bug, or is intended.

Using functional bean definition, if I remove the dependency spring-cloud-starter-function-webflux and try to deploy that into AWS Lambda, I get this error. If I keep that dependency, it runs fine.
I thought spring-cloud-starter-function-webflux dependency wasn't needed to run on AWS Lambda.

```
2021-12-15T10:59:14.101-03:00	13:59:14.092 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Searching for start class in manifest: jar:file:/var/runtime/lib/aws-lambda-java-core-1.2.0.jar!/META-INF/MANIFEST.MF

2021-12-15T10:59:14.102-03:00	Manifest-Version: 1.0

2021-12-15T10:59:14.102-03:00	Archiver-Version: Plexus Archiver

2021-12-15T10:59:14.102-03:00	Built-By: adsuresh

2021-12-15T10:59:14.102-03:00	Created-By: Apache Maven 3.5.0

2021-12-15T10:59:14.102-03:00	Build-Jdk: 1.8.0_144

2021-12-15T10:59:14.106-03:00	13:59:14.105 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Searching for start class in manifest: jar:file:/var/runtime/lib/aws-lambda-java-runtime-0.2.0.jar!/META-INF/MANIFEST.MF

2021-12-15T10:59:14.106-03:00	Manifest-Version: 1.0

2021-12-15T10:59:14.106-03:00	Archiver-Version: Plexus Archiver

2021-12-15T10:59:14.106-03:00	Built-By: root

2021-12-15T10:59:14.106-03:00	Created-By: Apache Maven 3.6.3

2021-12-15T10:59:14.106-03:00	Build-Jdk: 1.8.0_302

2021-12-15T10:59:14.107-03:00	13:59:14.107 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Searching for start class in manifest: jar:file:/var/runtime/lib/aws-lambda-java-serialization-0.2.0.jar!/META-INF/MANIFEST.MF

2021-12-15T10:59:14.107-03:00	Manifest-Version: 1.0

2021-12-15T10:59:14.107-03:00	Archiver-Version: Plexus Archiver

2021-12-15T10:59:14.107-03:00	Built-By: root

2021-12-15T10:59:14.107-03:00	Created-By: Apache Maven 3.6.3

2021-12-15T10:59:14.107-03:00	Build-Jdk: 1.8.0_302

2021-12-15T10:59:14.108-03:00	13:59:14.108 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Searching for start class in manifest: jar:file:/var/runtime/amzn-log4j-security-jdk11-0.1alpha.jar!/META-INF/MANIFEST.MF

2021-12-15T10:59:14.108-03:00	Manifest-Version: 1.0

2021-12-15T10:59:14.108-03:00	Premain-Class: Log4jPatch

2021-12-15T10:59:14.108-03:00	Created-By: 11.0.13 (Amazon.com Inc.)

2021-12-15T10:59:14.109-03:00	13:59:14.109 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Searching for start class in manifest: file:/var/task/META-INF/MANIFEST.MF

2021-12-15T10:59:14.110-03:00	Manifest-Version: 1.0

2021-12-15T10:59:14.110-03:00	Created-By: Maven Jar Plugin 3.2.0

2021-12-15T10:59:14.110-03:00	Build-Jdk-Spec: 11

2021-12-15T10:59:14.110-03:00	Implementation-Title: derived-strings-lambda

2021-12-15T10:59:14.110-03:00	Implementation-Version: 0.0.1-SNAPSHOT

2021-12-15T10:59:14.110-03:00	Main-Class: org.springframework.boot.loader.wrapper.ThinJarWrapper

2021-12-15T10:59:14.110-03:00	Start-Class: com.highroads.derivedstringslambda.DerivedStringsLambdaAppl

2021-12-15T10:59:14.110-03:00	ication

2021-12-15T10:59:14.111-03:00	Spring-Boot-Version: 2.6.1

2021-12-15T10:59:14.111-03:00	Spring-Boot-Classes: 

2021-12-15T10:59:14.141-03:00	13:59:14.140 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Loaded Start Class: class com.highroads.derivedstringslambda.DerivedStringsLambdaApplication

2021-12-15T10:59:14.141-03:00	13:59:14.141 [main] INFO org.springframework.cloud.function.utils.FunctionClassUtils - Main class: class com.highroads.derivedstringslambda.DerivedStringsLambdaApplication

2021-12-15T10:59:15.508-03:00	. ____ _ __ _ _

2021-12-15T10:59:15.508-03:00	/\\ / ___'_ __ _ _(_)_ __ __ _ \ \ \ \

2021-12-15T10:59:15.508-03:00	( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \

2021-12-15T10:59:15.508-03:00	\\/ ___)| |_)| | | | | || (_| | ) ) ) )

2021-12-15T10:59:15.508-03:00	' |____| .__|_| |_|_| |_\__, | / / / /

2021-12-15T10:59:15.508-03:00	=========|_|==============|___/=/_/_/_/

2021-12-15T10:59:15.527-03:00	:: Spring Boot ::

2021-12-15T10:59:16.109-03:00	2021-12-15 13:59:16.108 INFO 9 --- [ main] o.s.c.f.a.aws.CustomRuntimeInitializer : AWS Handler: org.springframework.cloud.function.adapter.aws.FunctionInvoker

2021-12-15T10:59:16.236-03:00	2021-12-15 13:59:16.236 ERROR 9 --- [ main] o.s.boot.SpringApplication : Application run failed

2021-12-15T10:59:16.236-03:00	java.lang.NoClassDefFoundError: org/springframework/cloud/function/web/source/DestinationResolver

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.adapter.aws.CustomRuntimeInitializer.initialize(CustomRuntimeInitializer.java:52) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.adapter.aws.CustomRuntimeInitializer.initialize(CustomRuntimeInitializer.java:34) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:610) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:378) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.boot.SpringApplication.run(SpringApplication.java:301) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.context.FunctionalSpringApplication.run(FunctionalSpringApplication.java:88) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.context.FunctionalSpringApplication.run(FunctionalSpringApplication.java:83) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.adapter.aws.FunctionInvoker.start(FunctionInvoker.java:126) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at org.springframework.cloud.function.adapter.aws.FunctionInvoker.<init>(FunctionInvoker.java:70) ~[task/:na]

2021-12-15T10:59:16.236-03:00	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]

2021-12-15T10:59:16.236-03:00	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:na]

2021-12-15T10:59:16.236-03:00	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[na:na]

2021-12-15T10:59:16.236-03:00	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source) ~[na:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader.newInstance(EventHandlerLoader.java:409) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader$StreamMethodRequestHandler.fromMethod(EventHandlerLoader.java:333) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader$StreamMethodRequestHandler.makeRequestHandler(EventHandlerLoader.java:346) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader.getThreeLengthHandler(EventHandlerLoader.java:680) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader.getHandlerFromOverload(EventHandlerLoader.java:699) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader.loadEventPojoHandler(EventHandlerLoader.java:811) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.EventHandlerLoader.loadEventHandler(EventHandlerLoader.java:613) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.AWSLambda.findUserMethods(AWSLambda.java:124) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.AWSLambda.startRuntime(AWSLambda.java:230) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.AWSLambda.startRuntime(AWSLambda.java:199) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at lambdainternal.AWSLambda.main(AWSLambda.java:193) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	Caused by: java.lang.ClassNotFoundException: org.springframework.cloud.function.web.source.DestinationResolver. Current classpath: file:/var/task/

2021-12-15T10:59:16.236-03:00	at lambdainternal.CustomerClassLoader.findClass(CustomerClassLoader.java:63) ~[aws-lambda-java-runtime-0.2.0.jar:na]

2021-12-15T10:59:16.236-03:00	at java.base/java.lang.ClassLoader.loadClass(Unknown Source) ~[na:na]

2021-12-15T10:59:16.236-03:00	at java.base/java.lang.ClassLoader.loadClass(Unknown Source) ~[na:na]

2021-12-15T10:59:16.236-03:00	... 24 common frames omitted

2021-12-15T10:59:16.240-03:00	org/springframework/cloud/function/web/source/DestinationResolver: java.lang.NoClassDefFoundError java.lang.NoClassDefFoundError: org/springframework/cloud/function/web/source/DestinationResolver at org.springframework.cloud.function.adapter.aws.CustomRuntimeInitializer.initialize(CustomRuntimeInitializer.java:52) at org.springframework.cloud.function.adapter.aws.CustomRuntimeInitializer.initialize(CustomRuntimeInitializer.java:34) at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:610) at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:378) at org.springframework.boot.SpringApplication.run(SpringApplication.java:301) at org.springframework.cloud.function.context.FunctionalSpringApplication.run(FunctionalSpringApplication.java:88) at org.springframework.cloud.function.context.FunctionalSpringApplication.run(FunctionalSpringApplication.java:83) at org.springframework.cloud.function.adapter.aws.FunctionInvoker.start(FunctionInvoker.java:126) at org.springframework.cloud.function.adapter.aws.FunctionInvoker.<init>(FunctionInvoker.java:70) at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source) Caused by: java.lang.ClassNotFoundException: org.springframework.cloud.function.web.source.DestinationResolver. Current classpath: file:/var/task/

```
**Sample**
my config is described on this ticket https://github.com/spring-cloud/spring-cloud-function/issues/783

**Comment by adolfo-cia** [comment]
Hi @olegz , mind checking on this when you have a min please? I just wanna make sure if I'm doing things right or it's a bug

**Comment by olegz** [comment]
I tested it before releasing 3.1.6 and actually had to fix something and it works, so this means there may be something that is missing hence my inability to reproduce. So the best way is for you to create a small project that reproduces the issue and push it to GotHub so we can clone, run and see what the issue is

**Comment by adolfo-cia** [comment]
Thx for the reply.

I believe this was happening with one of the aws sample project in here. Anyway, lemme check that out or get a project in github and come back to you. Thanks again.

**Comment by adolfo-cia** [comment]
@olegz I confirm that the sample project in this project, function-functional-sample-aws, has the same issue.

-  use the 3.1.6 version
- remove/comment spring-cloud-starter-function-webflux dependency
- uncomment this line in the main class:
       `FunctionalSpringApplication.run(FunctionConfiguration.class, args);`

**Comment by adolfo-cia** [comment]
@olegz sorry to bother you, any chance you reviewed last comment?

**Comment by olegz** [comment]
We're on holidays, i'll try to look if i get a chance otherwise will look after the holidays

**Comment by adolfo-cia** [comment]
thx!

**Comment by olegz** [comment]
@adolfo-cia So, finally I managed to take a look and here is the deal.
Based on your stack trace i see that you are using Custom Runtime. . . That doesn't actually mean that you do, but there was a bug that would bootstrap it anyway. It was fixed, which means you are using some older 3.1.x version. 
Consider moving to 3.2.1.
That said, the `DestinationResolver` is no longer necessary. In fact that whole block of code (the one that triggered the exception) is no longer necessary, so I am removing it as part of this issue.
